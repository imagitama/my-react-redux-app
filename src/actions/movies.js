export const FETCH_MOVIES = 'FETCH_MOVIES'
export function fetchMovies() {
  return {
    type: FETCH_MOVIES
  }
}

export const MOVIES_FETCHED = 'MOVIES_FETCHED'
export function moviesFetched({ movies }) {
  return {
    type: MOVIES_FETCHED,
    movies
  }
}

export const FAILED_TO_FETCH_MOVIES = 'FAILED_TO_FETCH_MOVIES'
export function failedToFetchMovies({ error }) {
  return {
    type: FAILED_TO_FETCH_MOVIES,
    errorMessage: error.message
  }
}