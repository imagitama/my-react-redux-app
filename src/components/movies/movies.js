import React from 'react'
import { connect } from 'react-redux'
import styled from 'react-emotion'
import { Movie } from '../movie/movie'
import { LoadingIndicator } from '../loading-indicator/loading-indicator'

const StyledMovies = styled('ul')({
  margin: 0,
  padding: 0
})

export const Movies = connect(state => state.movies)(({ fetching, movies }) => 
  fetching ? 
    <LoadingIndicator /> :
    <StyledMovies>
      {movies.map(movie => <Movie key={movie.id} {...movie} />)}
    </StyledMovies>
)