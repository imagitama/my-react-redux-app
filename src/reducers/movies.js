export function moviesReducer(state = { fetching: false, movies: [], errorMessage: '' }, action = { type: '' }) {
  switch (action.type) {

    // Tried using constants for each one but Jest doesn't like that :(
    case 'MOVIES_FETCH':
      if (state.fetching) {
        return state
      }

      return {
        ...state,
        fetching: true
      }

    case 'MOVIES_FETCH_SUCCESS':
      return {
        ...state,
        fetching: false,
        errorMessage: '',
        movies: action.movies
      }

    case 'MOVIES_FETCH_FAILED':
      return {
        ...state,
        fetching: false,
        errorMessage: action.errorMessage
      }

    default:
      return state
  }
}