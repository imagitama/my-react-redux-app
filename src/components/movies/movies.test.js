import React from 'react'
import { mount } from 'enzyme'
import { Movies } from './movies'

it('Matches the snapshot', () => {
  expect(mount(<Movies />)).toMatchSnapshot()
})