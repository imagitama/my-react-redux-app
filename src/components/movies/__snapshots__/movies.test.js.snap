// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<Component
  movies={Array []}
>
  <Styled(ul)>
    <ul
      className="css-1l95nvm"
    />
  </Styled(ul)>
</Component>
`;
